migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/sql_schema_differ_flavour/mssql.rs

```diff
@@ -39,20 +42,1086 @@ impl SqlSchemaDifferFlavour for MssqlFlavour {
     }
 
     fn column_type_change(&self, differ: &ColumnDiffer<'_>) -> Option<ColumnTypeChange> {
-        if differ.previous.column_type_family() == differ.next.column_type_family() {
-            None
+        let previous_family = differ.previous.column_type_family();
+        let next_family = differ.next.column_type_family();
+
+        let family_comparison = || {
+            if previous_family == next_family {
+                None
+            } else {
+                Some(family_change_riskyness(previous_family, next_family))
+            }
+        };
+
+        if self.features().contains(MigrationFeature::NativeTypes) {
+            let previous_type: Option<MsSqlType> = differ.previous.column_native_type();
+            let next_type: Option<MsSqlType> = differ.next.column_native_type();
+
+            match (previous_type, next_type) {
+                (None, _) | (_, None) => family_comparison(),
+                (Some(previous), Some(next)) if previous == next => None,
+                (Some(previous), Some(next)) => Some(native_type_change_riskyness(previous, next)),
+            }
         } else {
-            Some(type_change_riskyness(differ))
+            family_comparison()
         }
     }
 }
 
-fn type_change_riskyness(differ: &ColumnDiffer<'_>) -> ColumnTypeChange {
-    match (differ.previous.column_type_family(), differ.next.column_type_family()) {
+fn family_change_riskyness(previous: &ColumnTypeFamily, next: &ColumnTypeFamily) -> ColumnTypeChange {
+    match (previous, next) {
         (_, ColumnTypeFamily::String) => ColumnTypeChange::SafeCast,
         (ColumnTypeFamily::String, ColumnTypeFamily::Int)
         | (ColumnTypeFamily::DateTime, ColumnTypeFamily::Float)
         | (ColumnTypeFamily::String, ColumnTypeFamily::Float) => ColumnTypeChange::NotCastable,
         (_, _) => ColumnTypeChange::RiskyCast,
     }
 }
+
+fn native_type_change_riskyness(previous: MsSqlType, next: MsSqlType) -> ColumnTypeChange {
+    use ColumnTypeChange::*;
+    use MsSqlTypeParameter::*;
+
```

Comment:
Should we use a `MsSQLType::*` to reduce visual noise by getting rid of all the `MsSqlType::` below?